driver-core/src/main/java/com/datastax/driver/core/SessionManager.java

```diff
@@ -611,6 +615,21 @@ void cleanupIdleConnections(long now) {
         }
     }
 
+    @Override
+    protected void checkNotInEventLoop() {
+        if (!CHECK_IO_DEADLOCKS)
+            return;
+        for (EventExecutor executor : cluster.manager.connectionFactory.eventLoopGroup) {
+            if (executor.inEventLoop()) {
+                throw new IllegalStateException(
+                        "Detected a synchronous Session call (execute() or prepare()) on an I/O thread, " +
```

Comment:
"This is a bug" could be interpreted as a bug in the driver code by users who don't understand the issue.  Can we remove that statement, i.e. instead: 

> Detected a synchronous Session call (execute() or prepare()) on an I/O thread, this can cause deadlocks or unpredictable behavior. Make sure your Future callbacks only use async calls, or schedule them on a different executor.